src/perform_step/composite_perform_step.jl

```diff
@@ -51,10 +56,25 @@ function initialize!(integrator, cache::CompositeCache{Tuple{T1, T2}, F}) where
         initialize!(integrator, @inbounds(cache.caches[1]))
     elseif cache.current == 2
         initialize!(integrator, @inbounds(cache.caches[2]))
+        reset_alg_dependent_opts!(integrator.opts.controller, integrator.alg.algs[1],
+            integrator.alg.algs[2])
     end
     resize!(integrator.k, integrator.kshortsize)
 end
 
+"""
+If the user mixes adaptive and non-adaptive algorithms then, right after
+initialize!, make integrator.opts match the default adaptivity such that
+the behaviour is consistent.
+In particular, prevents dt ⟶ 0 if starting with non-adaptive alg and opts.adaptive=true,
+and dt=cst if starting with adaptive alg and opts.adaptive=false.
+"""
```

Comment:
It would be nice to add a test for this. I can't say I've ever tried a non-adaptive alg in an auto-switch. When did this come up?

Reply:
This came up while testing to see if the ODESolution issue was fixed...